### PR TITLE
Implementation of combined Ability and StatusEffect system with examples

### PR DESCRIPTION
**Pull request by logatsang**
Summary of changes:
- created and implemented a facade (CharacterStatusEffectFacade) to handle the application of StatusEffects to GameCharacters during combat and moved a number of functions there
- implemented a number of test Abilities and StatusEffects (details below)
- minor style adjustments

This will run, but abilities cannot be used due to a NullPointerException on line 47 of CombatInputListener.java: savedAbility is not initialized, which causes the error. I don't know how to fix this and I'm too tired right now to read the rest of the class.

On how StatusEffects are applied:
 - specific changes in character values (referred to internally as "subeffects") are recorded in a List of Strings in the StatusEffect object 
 - when a StatusEffect is applied to a GameCharacter during combat, the CharacterStatusEffectFacade class parses and applies each subeffect

The strings are not an exceptionally pretty solution, but they will do for now.

### CODE REVIEW

File: src/main/java/combat_system/CharacterStatusEffectFacade.java
Comment (edwardsues) [inline src/main/java/combat_system/CharacterStatusEffectFacade.java:60]: perhaps make the split of the subeffect a function? it looks like you use it multiple times.

**Review by edwardsues — CHANGES_REQUESTED** [review]
everything else looks good!, just some minor refactoring that could be done

File: src/main/java/combat_system/CharacterStatusEffectFacade.java
Comment (logatsang) [reply src/main/java/combat_system/CharacterStatusEffectFacade.java]: Good idea! In my latest commit, I have relegated the functionality of splitting the subeffect definition string to two separate methods.

**Review by edwardsues — APPROVED** [review]
I'd say wait for 1 more person to review the pull request before merging.

File: src/main/java/characters/Ability.java
Comment (wBernie) [inline src/main/java/characters/Ability.java:5]: doesn't seem like we need this import anymore

**Review by wBernie — APPROVED** [review]

**Pull request merged**
